library/testing/TestSDKCamera.cxx

```diff
@@ -0,0 +1,230 @@
+#include <camera.h>
+#include <engine.h>
+#include <loader.h>
+#include <window.h>
+
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+
+bool compareDouble(double a, double b)
+{
+  return std::fabs(a - b) < 128 * std::numeric_limits<double>::epsilon();
+}
+
+bool compareVec(f3d::camera::vector3_t vec1, f3d::camera::vector3_t vec2)
+{
+  return compareDouble(vec1[0], vec2[0]) && compareDouble(vec1[1], vec2[1]) &&
+    compareDouble(vec1[2], vec2[2]);
+}
+
+bool compareMat(f3d::camera::matrix4_t mat1, f3d::camera::matrix4_t mat2)
+{
+  bool ret = true;
+  for (size_t i = 0; i < mat1.size(); i++)
+  {
+    ret &= compareDouble(mat1[i], mat2[i]);
+  }
+  return ret;
```

Comment:
I don't like `&=`, and you can actually early exit in case of failure.

```suggestion
  for (size_t i = 0; i < mat1.size(); i++)
  {
    if (!compareDouble(mat1[i], mat2[i]))
    {
      return false;
    }
  }
  return true;
```